types/graphql-depth-limit/index.d.ts

```diff
@@ -0,0 +1,20 @@
+// Type definitions for graphql-depth-limit 1.1
+// Project: https://github.com/stems/graphql-depth-limit#readme
+// Definitions by: Siim Tiilen <https://github.com/eritikass>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.8
+
+declare var d: GraphqlDepthLimit.depthLimit;
```

Comment:
This could be written as:
```ts
declare function depthLimit(depthLimit: number, options?: depthLimit.Options, callback?: (obj: any) => void): any;
export = depthLimit;

declare namespace depthLimit {
    interface Options {
        ignore: Array<string | RegExp | ((queryDepths: any[]) => boolean)>;
    }
}
```

Also, could there be a more specific return type?

Reply:
tnx, updated - 5fbb92cbbb
---- 
about the return type sadly not, offical graphql types says that validate rules can be anything

> https://github.com/DefinitelyTyped/DefinitelyTyped/blob/0c23284637983455f4aff76ae58c54b11b1a9974/types/graphql/validation/validate.d.ts#L25